src/devices/cpu/tlcs900/tmp94c241.cpp

```diff
@@ -0,0 +1,1769 @@
+{
+	return m_t01mod;
+}
+
+void tmp94c241_device::t01mod_w(uint8_t data)
+{
+	m_t01mod = data;
+}
+
+uint8_t tmp94c241_device::tffcr_r()
+{
+	return m_tffcr;
+}
+
+enum
+{
+	FF_INVERT,
+	FF_SET,
+	FF_CLEAR,
+	FF_DONTCARE
+};
+
+void tmp94c241_device::change_timer_flipflop(uint8_t flipflop, uint8_t operation)
+{
+	/* First we update the timer flip-flop */
+	bool* ff_state = &m_timer_flipflops[flipflop];
+
+	switch(operation)
+	{
+	case FF_INVERT:
+		*ff_state = !(*ff_state);
+		break;
+	case FF_SET:
+		*ff_state = true;
+		break;
+	case FF_CLEAR:
+		*ff_state = false;
+		break;
+	default:
+		// invalid operation
+		return;
+	}
+
+	/* The value of the flipflop is only exposed to a pin in certain modes of operation
+	   determined by fields of the port function registers.
+
+	   So here we bail out if the flipflop is not routed to its corresponding port bit:
+	*/
+	switch( flipflop )
+		{
+		case 0x1:
+			if (!BIT(m_port_function[PORT_C], 0) || BIT(m_port_control[PORT_C], 0)) return;
+			break;
+		case 0x7:
+			if (!BIT(m_port_function[PORT_C], 0) || !BIT(m_port_control[PORT_C], 0)) return;
+			break;
+		case 0x3:
+			if (!BIT(m_port_function[PORT_C], 1) || BIT(m_port_control[PORT_C], 1)) return;
+			break;
+		case 0xb:
+			if (!BIT(m_port_function[PORT_C], 1) || !BIT(m_port_control[PORT_C], 1)) return;
+			break;
+		case 0x4:
+			if (!BIT(m_port_function[PORT_D], 0)) return;
+			break;
+		case 0x6:
+			if (!BIT(m_port_function[PORT_D], 4)) return;
+			break;
+		case 0x8:
+			if (!BIT(m_port_function[PORT_E], 0)) return;
+			break;
+		case 0xA:
+			if (!BIT(m_port_function[PORT_E], 4)) return;
+			break;
+		default:
+			// invalid flip flop
+			return;
+		}
+
+	/* And here we actually send the value to the corresponding pin */
+	uint8_t new_port_value = 0;
+	switch( flipflop )
+		{
+		case 0x1:
+		case 0x7:
+			new_port_value = m_port_latch[PORT_C] & 0xfe;
+			if (*ff_state) new_port_value |= 0x01;
+			port_w<PORT_C>(new_port_value);
+			break;
+		case 0x3:
+		case 0xb:
+			new_port_value = m_port_latch[PORT_C] & 0xfd;
+			if (*ff_state) new_port_value |= 0x02;
+			port_w<PORT_C>(new_port_value);
+			break;
+		case 0x4:
+			new_port_value = m_port_latch[PORT_D] & 0xfe;
+			if (*ff_state) new_port_value |= 0x01;
+			port_w<PORT_D>(new_port_value);
+			break;
+		case 0x6:
+			new_port_value = m_port_latch[PORT_D] & 0xef;
+			if (*ff_state) new_port_value |= 0x10;
+			port_w<PORT_D>(new_port_value);
+			break;
+		case 0x8:
+			new_port_value = m_port_latch[PORT_E] & 0xfe;
+			if (*ff_state) new_port_value |= 0x01;
+			port_w<PORT_E>(new_port_value);
+			break;
+		case 0xa:
+			new_port_value = m_port_latch[PORT_E] & 0xef;
+			if (*ff_state) new_port_value |= 0x10;
+			port_w<PORT_E>(new_port_value);
+			break;
+		}
+}
+
+void tmp94c241_device::tffcr_w(uint8_t data)
+{
+	change_timer_flipflop( 1, (data >> 2) & 3 );
+	change_timer_flipflop( 3, (data >> 6) & 3 );
+	m_tffcr = data | 0xcc;
+}
+
+void tmp94c241_device::treg23_w(offs_t offset, uint8_t data)
```

Comment:
why isn't that two independant handlers?

Reply:
yes, you're right! Some timers SFRs pack together data for a couple timers, but that's not this case for these ones. I'll fix this!

![Screenshot From 2025-02-09 18-46-55](https://github.com/user-attachments/assets/97f96190-c446-4a78-8ae4-f5bd63d1bec8)

Reply:
I think I fixed this one as well.